### PR TITLE
Add C++ Solution for Convention II (Silver)

### PR DESCRIPTION
**Pull request by ryanchou-dev**
i was going to do python, but it's really bad for problems like this

ill start working on the two pointers quiz soon

---
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [x] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

File: solutions/usaco-859.mdx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	});
+
+	int time = 0, cur = 0, ans = 0;
+
+	// sorted by priority.
+	set<vector<int>> waiting;
+
+	// as long as we haven't processed all cows or there are still cows waiting.
+	while (cur < n || waiting.size() > 0) {
+
```
Comment (maggie-j-liu) [inline solutions/usaco-859.mdx:52]: is this empty line intentional?

**Review by maggie-j-liu — APPROVED** [review]

File: solutions/usaco-859.mdx (hunk 1 above)
Comment (ryanchou-dev) [reply solutions/usaco-859.mdx:52]: yeah, for the comment. should i remove it?

Comment (maggie-j-liu) [reply solutions/usaco-859.mdx:52]: nah i was just wondering

File: solutions/usaco-859.mdx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		cows.push_back({i, start, duration});
+	}
+	// sort by arrival time.
+	sort(cows.begin(), cows.end(), [](const vector<int> &a, const vector<int> &b) {
+		return a[1] < b[1];
+	});
+
+	int time = 0, cur = 0, ans = 0;
+	set<vector<int>> waiting;  // sorted by priority
+	// as long as we haven't processed all cows or there are still cows waiting.
```
Comment (nwatx) [inline solutions/usaco-859.mdx]:
```suggestion
	// as long as we haven't processed all cows or there are still cows waiting
```

**Review by nwatx — APPROVED** [review]

File: solutions/usaco-859.mdx (hunk 1 above)
Comment (ryanchou-dev) [reply solutions/usaco-859.mdx:48]: image

Comment (SansPapyrus683) [reply solutions/usaco-859.mdx:48]: what

Comment (ryanchou-dev) [reply solutions/usaco-859.mdx:48]: image

Comment (SansPapyrus683) [reply solutions/usaco-859.mdx:48]: ...you don't need that whitespace for that comment.

**Review by SansPapyrus683 — APPROVED** [review]
not sure if i approved alr

File: solutions/usaco-859.mdx [hunk 3]
```diff
@@ -0,0 +1,71 @@
+---
+id: usaco-859
+source: USACO Silver 2018 December
+title: Convention II
+author: Ryan Chou
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_convention2_silver_dec18.html)
+
+Due to the low bounds on $N$, we can naively simulate the events.
```
Comment (caoash) [inline solutions/usaco-859.mdx:10]: maybe explain why you sort by arrival time

File: solutions/usaco-859.mdx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+		cin >> start >> duration;
+		cows.push_back({i, start, duration});
+	}
+	// sort by arrival time.
+	sort(cows.begin(), cows.end(), [](const vector<int> &a, const vector<int> &b) {
+		return a[1] < b[1];
+	});
+
+	int time = 0, cur = 0, ans = 0;
+	set<vector<int>> waiting;  // sorted by priority
```
Comment (caoash) [inline solutions/usaco-859.mdx:45]: explain why you sort by priority here

**Review by caoash — CHANGES_REQUESTED** [review]

**Review by caoash — CHANGES_REQUESTED** [review]
changes

File: solutions/usaco-859.mdx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+	int n;
+	vector<vector<int>> cows;
+	cin >> n;
+	for (int i = 0; i < n; i++) {
+		int start, duration;
+		cin >> start >> duration;
+		cows.push_back({i, start, duration});
+	}
+	// sort by arrival time.
+	sort(cows.begin(), cows.end(), [](const vector<int> &a, const vector<int> &b) {
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-859.mdx:40]: RIP over 80 characters

**Review by Aadit-Ambadkar — CHANGES_REQUESTED** [review]

File: solutions/usaco-859.mdx (hunk 5 above)
Comment (ryanchou-dev) [reply solutions/usaco-859.mdx:40]: image
 
👀

**Pull request merged**